pkg/volume/csi/csi_plugin.go

```diff
@@ -392,10 +381,11 @@ func (p *csiPlugin) NewMounter(
 		return nil, err
 	}
 
-	// TODO(pohly): check CSIDriver.Spec.Mode to ensure that the CSI driver
+	// Check CSIDriver.Spec.Mode to ensure that the CSI driver
```

Comment:
should this check be feature gated on CSIInlineVolume, since the volume mode field is controlled by that feature gate?

Reply:
That won't change much because the result "should" be the same also when the feature is disabled. But to avoid potential mistakes, I short-circuited the logic in `supportsVolumeMode`:
```
func (p *csiPlugin) supportsVolumeMode(driver string, volumeMode storage.CSIVolumeMode) error {
	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) {
		// Nothing to check, feature disabled.
		return nil
	}
```